client/prometheus.go

```diff
@@ -0,0 +1,225 @@
+package client
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/hellofresh/stats-go/bucket"
+	"github.com/hellofresh/stats-go/incrementer"
+	"github.com/hellofresh/stats-go/state"
+	"github.com/hellofresh/stats-go/timer"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Prometheus is Client implementation for prometheus
+type Prometheus struct {
+	sync.Mutex
+
+	unicode            bool
+	httpMetricCallback bucket.HTTPMetricNameAlterCallback
+	httpRequestSection string
+
+	namespace  string
+	increments map[string]*incrementer.Prometheus
+	states     map[string]*state.Prometheus
+	labels     map[string]string
+}
+
+// NewPrometheus builds and returns new Prometheus instance
+func NewPrometheus(namespace string) (*Prometheus, error) {
+	client := &Prometheus{
+		namespace:  namespace,
+		increments: make(map[string]*incrementer.Prometheus),
+		states:     make(map[string]*state.Prometheus),
+		labels:     make(map[string]string),
+	}
+	return client, nil
+}
+
+// WithLabels adds labels map to metrics
+func (c *Prometheus) WithLabels(labels map[string]string) Client {
+	c.Lock()
+	defer c.Unlock()
+
+	c.labels = labels
```

Comment:
This will not work. Imagine you have the following code:

```go
client, _ := stats.NewClient("prometheus://your_namespace")

// HTTP-handler that handles one request
go func() {
    client.WithLabels(map[string]string{"foo": "bar"}).TrackMetric("a1", operation)
}()

// HTTP-handler that handles another request
go func() {
    client.WithLabels(map[string]string{"baz": "qux"}).TrackMetric("a2", operation)
}()
```

in this case there is no guarantee which labels will be assigned to any of the metric - all `Track*` methods do not have any side-effects, they get some parameters, make some operations with the data and finish. This method sets global state for an object that is used in `Track*`. This is why I was suggecting breaking backward compatibility and changing `bucket.MetricOperation`.